Add tutorial

Lestropie opened the pull request:
Builds upon #28 since the tutorial uses a couple of functionalities added there.

- [x] Check for consistency of formatting across all documents.

- [x] Add some images / screenshots:

    - [x] Appearance of spin-echo EPIs and what they look like when phase encoding reversal is not properly achieved
    - [x] Navigating the protocol PDF printouts

- [x] Reproduce steps independently once data have been successfully cloned to OSF (https://osf.io/dz5mc/).
    **edit**: Changed data location from OSF to GitHub via submodule: https://github.com/Australian-Epilepsy-Project/ProtocolDeviations

- [x] Check that all image paths are correct and get embedded into Markdown correctly.

I have constructed protocol cards for a larger number of sequence parameter modifications, partly as a torture test looking for sequence parameters that the software is incapable of detecting. But I would prefer to prioritise getting this merged prior to the coming workshop; additional testing / tutorial datasets can be added after the fact.

Now if you'll excuse me, I'm-a go sleep for a bit:

<img width="619" height="52" alt="Screenshot from 2025-08-02 15-33-16" src="https://github.com/user-attachments/assets/0449f152-810e-4344-8a2b-48b23954c306" />

Lestropie committed:
Various contributions arising from workshop content generation

- New comparison "absent" enforces that a metadata key must not appear in the input data. This must be accompanied by directive "compulsory": false (see below).
- Update documentation to reflect breadth of private metadata fields loaded by the software.
- Perform various checks of the formatting of an input template, in order to provide more interpretable error messages to those attempting to build their own templates.
- Add Siemens private field encoding diffusion MRI b-value to list of private fields extracted.
- New template field "compulsory": setting this to "false" for a metadata field results in preserving the establishment of a match between template and input data if that field is absent from the input data; instead it is only considered a mismatch if the field is present but has a different value to that of the template.

Lestropie committed:
Docs: First commit of tutorial

Lestropie committed:
Add submodule for ProtocolDeviations repo

Also updates docs accordingly.

Lestropie committed:
Updates to workshop content

- Pull updated git submodule data.
- Upload images: highlighting PDF printout content, showing some images in MRtrix3 mrview.
- Harmonise some formatting of shell snippets vs. stdout contents.
- Remove use of markdown "details" functionality (made sense when the tutorial was one long document; but spread across many documents as it is, showing all content makes more sense).

Lestropie commented:
Changed PR base to `dev` so that `dev` can separately be PR'd to master accompanied by a version increment.

aaroncapon reviewed (approved):
Looks really good to me!

Lestropie committed:
Docs: Tutorial fixes

- Standardise reference to xpdf software for reproduction of pdftotext command calls.
- Fix some image paths.
- Remove ellipses from JSON snippets as they are highlighted on GitHub as malformed.
